hs/src/Reach/Connector/ALGO.hs

```diff
@@ -2032,6 +2032,21 @@ ce = \case
     ca oa
     cextract start sz
     cfrombs t
+  DLE_ObjectSet at obj_a fieldName val_a -> do
+    -- This makes the (unsound?) assumption that when adding a new field to an object,
+    -- all copied fields should stay in the order they were already in.
+    let val_t = argTypeOf val_a
+    let oldFieldTypes = argObjstrTypes obj_a
+    let newFieldTypes = M.toAscList $ M.insert fieldName val_t $ M.fromList oldFieldTypes
```

Comment:
These two things should be exactly the same

Reply:
Not if we're adding a new field or changing the field's type

Reply:
We don't want to support that